Python/pylifecycle.c

```diff
@@ -2719,6 +2729,73 @@ init_sys_streams(PyThreadState *tstate)
 }
 
 
+#ifdef __ANDROID__
+#include <android/log.h>
+
+static PyObject *
+android_log_write_impl(PyObject *self, PyObject *args)
+{
+    int prio = 0;
+    const char *tag = NULL;
+    const char *text = NULL;
+    if (!PyArg_ParseTuple(args, "isy", &prio, &tag, &text)) {
+        return NULL;
+    }
+
+    // Despite its name, this function is part of the public API
+    // (https://developer.android.com/ndk/reference/group/logging).
+    __android_log_write(prio, tag, text);
+    Py_RETURN_NONE;
+}
+
+
+static PyMethodDef android_log_write_method = {
+    "android_log_write", android_log_write_impl, METH_VARARGS
```

Comment:
Would it be possible to use METH_FASTCALL?

Reply:
You can replace PyArg_ParseTuple() with _PyArg_ParseStack().

Reply:
I do not think that it will make significant difference in this case. It is better to use the stable public API.

Reply:
I agree: this isn't a performance-critical area.

Reply:
Ok, let's go with VARARGS.